### DISCUSSION TITLE
writeHeapSnapshot is not a function

### DISCUSSION THREAD

**Discussion by cqh963852** [vitest-dev/vitest#4389] [Q&A] [answered]
I am creating a frontend library.

I want to be able to analyze memory snapshots in a test program.

```
import { writeHeapSnapshot } from "v8";
```

It can be executed normally without specifying environment.

But my program requires some document-related operations. So I added the configuration as follows

```
import { defineConfig } from "vitest/config";

export default defineConfig({
  test: {
   // use jsdom
    environment: "jsdom",
  },
});
```

![image](https://github.com/vitest-dev/vitest/assets/17702287/54468802-8bbb-46dc-b4b4-a7295696d5c8)

It will give me an error.

What should I do to make dom operations and v8 snapshot work at the same time?

**Comment by cqh963852** [comment]
I found that, If change

```
import { writeHeapSnapshot } from "v8";
```

to

```
const { writeHeapSnapshot } = require("v8");
```

It can works.

The objects to the console are as follows.
This situation seems so strange.
![image](https://github.com/vitest-dev/vitest/assets/17702287/3481d15e-1f42-4471-892b-798faac5b67a)
![image](https://github.com/vitest-dev/vitest/assets/17702287/800bb6f6-68f2-4b1f-95f5-d9be226eaa18)

**Comment by AriPerkkio** [comment]
What version is that on? Tested with `0.34.6` and `1.0.0-beta-3` and it works fine:
![image](https://github.com/vitest-dev/vitest/assets/14806298/1264f8f6-880e-401e-98c2-955eabf6b7ae)

**Reply by cqh963852** [reply]
I am using yarn 4

```
"v8": "^0.1.0",
"vitest": "^0.32.2"
```

After update to "0.34.6" still has same problem

![image](https://github.com/vitest-dev/vitest/assets/17702287/b12a7d07-2ae3-40b0-b61b-d6459057ec79)

![image](https://github.com/vitest-dev/vitest/assets/17702287/e94faecc-97a5-43cd-9650-5e13cf6331f3)


If using pnpm + vitest 1.0.0-beta.3 still has sam problem

![image](https://github.com/vitest-dev/vitest/assets/17702287/f851bd98-7406-493a-8665-217a20c85354)

**Reply by AriPerkkio** [reply]
> I am using yarn 4
>
> ```
> "v8": "^0.1.0",
> "vitest": "^0.32.2"
> ```

Is it intentionally using `v8` NPM package and not the NodeJS API? So https://www.npmjs.com/package/v8?activeTab=code and not https://nodejs.org/api/v8.html#v8writeheapsnapshotfilenameoptions.

**Reply by AriPerkkio** [reply, marked as answer]
So just run `yarn remove v8` or do following change:

```diff
-import { writeHeapSnapshot } from "v8";
+import { writeHeapSnapshot } from "node:v8";
```

**Reply by cqh963852** [reply]
Thank you very much, It helps. 

> Is it intentionally using v8 NPM package and not the NodeJS API?

It was my first time using it, I didn’t notice `node:v8`

**Comment by cqh963852** [comment]
I used github ci to execute this test, and the same error occurred.

![image](https://github.com/vitest-dev/vitest/assets/17702287/b710db00-b3d4-4146-9597-3230e30ac290)

https://github.com/idealjs/reactive/actions/runs/6684588844/job/18162003995

**Comment by sheremet-va** [comment]
This is the same issue as https://github.com/vitest-dev/vitest/issues/4350

For some reason, Vitest imports from `__vite_browser:vm` instead of `vm`
